lib/worldwide/deprecated_time_zone_mapper.rb

```diff
@@ -8,6 +10,103 @@ class << self
       end
+
+      # Map IANA timezones to Rails timezones
+      def to_rails(iana_timezone)
+        # If the timezone is already a supported Ruby timezone, return it
+        return iana_timezone if RAILS_TIMEZONES.key?(iana_timezone&.to_sym)
+
+        # IANA timezones has aliases. If the timezone is an alias, return the corresponding IANA timezone
+        timezone_alias = TIMEZONES_ALIASES.find do |timezone, aliases|
+          return timezone.to_s if aliases.include?(iana_timezone)
+        end
+
+        # If the alias is also a supported Rails timezone, return it
+        return timezone_alias if RAILS_TIMEZONES.key?(timezone_alias&.to_sym)
+
+        timezone = timezone_alias || iana_timezone
+
+        matching_timezones = find_matching_timezones(timezone)
+
+        return timezone if matching_timezones.empty?
+
+        iana_tz_info = TZInfo::Timezone.get(timezone)
+
+        # Find the closest Rails timezone based on geographical coordinates
+        closest_timezone = find_closest_timezone(iana_tz_info, matching_timezones)
+
+        closest_timezone
+      end
+
+      private
+
+      def extract_offset(offset)
+        match = offset.match(/\(GMT([+-]\d{2}:\d{2})\)/)
+        match ? match[1] : nil
+      end
+
+      # Get the Rails timezones that have the same GMT offset as the given IANA timezone
+      def find_matching_timezones(timezone)
+        return [] if ActiveSupport::TimeZone[timezone].nil?
+
+        iana_gmt_offset = ActiveSupport::TimeZone[timezone].formatted_offset
+
+        RAILS_TIMEZONES.filter do |_, offset|
+          ruby_offset = extract_offset(offset)
+          iana_gmt_offset == ruby_offset
+        end
+      end
+
+      def find_closest_timezone(timezone, neighboring_timezones)
+        # Get the geographical coordinates for the given timezone
+        coordinates = get_coordinates(timezone)
+
+        return timezone if coordinates.nil?
+
+        # Iterate through the Rails timezones to find the closest match
+        closest_timezone = nil
+        shortest_distance = Float::INFINITY
+
+        neighboring_timezones.keys.each do |neighboring_timezone|
+          timezone_info = TZInfo::Timezone.get(neighboring_timezone.to_s)
+          timezone_coordinates = get_coordinates(timezone_info)
+
+          if timezone_coordinates.nil?
+            break
+          end
+
+          distance = Geokit::LatLng.distance_between(
+            coordinates,
+            timezone_coordinates,
+          )
+
+          if distance < shortest_distance
+            shortest_distance = distance
+            closest_timezone = neighboring_timezone.to_s
+          end
+        end
+
+        closest_timezone
+      end
+
+      def get_coordinates(timezone)
+        # Get the timezone identifier
+        identifier = timezone.identifier
+
+        # Use the TZInfo::Country class to find the country and its timezones
+        TZInfo::Country.all.each do |country|
```

Comment:
This is a somewhat expensive scan to perform `1 + neighboring_zones` times on every lookup. In my review comments, I will recommend that we precompute this information, either in a kind of initializer or even as a gem build artifact (a yaml file).